datastore/mongo/mongo.go

```diff
@@ -113,6 +113,7 @@ func (c *Client) ensureMongoIndices() {
 	c.ensureIndex(EventCollection, "event_type", false, nil)
 	c.ensureIndex(EventCollection, "app_metadata.uid", false, nil)
 	c.ensureIndex(AppCollections, "group_id", false, nil)
+	c.ensureIndex(AppCollections, "title", true, nil)
```

Comment:
I think this index should also be scoped to the `group_id` along with the `document_status`. That way multiple apps under different groups can still have the same name, and then we can still reuse the same name for an app that has been deleted.

Reply:
> In hindsight, since we are now calling IsAppTitleUnique i will be removing the unique index

Reply:
Please let's use the index. Indexes are the last line of defence for data issues. I prefer Indexes over software checks. The actual advantage software checks give us over indexes is human-readable messages, but we can transform index errors into human-readable messages with a little more work.

Reply:
That would make the behaviors of our database layers uneven - badger doesn't have unique indexes.
But yes we can extract the error details using `mongo.IsDuplicateKeyError` and [this](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/errors.go#L276:6), the only problem here is the details would be awkward to return to the frontend.

We can leave the software check & keep the index for mongo, from the service layer perspective the behaviors are still even.

Reply:
@danvixent This is fine. Let's keep both for mongo & only software checks for badger. Realise also that we don't currently recommend badger for prod but people are running mongo in prod so we need to be more careful.

Reply:
That works 👍